pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -928,14 +960,25 @@ func (vs *VSphere) HasClusterID() bool {
 	return true
 }
 
-// Notification handler when node is registered.
-func (vs *VSphere) NodeRegistered(node *v1.Node) {
-	glog.V(4).Infof("Node Registered: %+v", node)
+// Notification handler when node is added into k8s cluster.
+func (vs *VSphere) NodeAdded(obj interface{}) {
+	node, ok := obj.(*v1.Node)
+	if node == nil || !ok {
+		return
+	}
+
+	glog.V(4).Infof("Node added: %+v", node)
 	vs.nodeManager.RegisterNode(node)
 }
 
-// Notification handler when node is unregistered.
-func (vs *VSphere) NodeUnregistered(node *v1.Node) {
-	glog.V(4).Infof("Node Unregistered: %+v", node)
+// Notification handler when node is removed from k8s cluster.
+func (vs *VSphere) NodeDeleted(obj interface{}) {
+	node, ok := obj.(*v1.Node)
+	if node == nil || !ok {
+		return
+	}
+
+	glog.V(4).Infof("Node deleted: %+v", node)
 	vs.nodeManager.UnRegisterNode(node)
```

Comment:
How did you test this node deletion notification? I dont see it in the testing done section.

Reply:
Tested and updated Testing Done section.